config-linux.md

```diff
@@ -685,6 +699,48 @@ The following parameters can be specified to set up seccomp:
 }
 ```
 
+### <a name="seccompstate" />The Seccomp State
+The seccomp state is a data structure passed via a UNIX socket.
+The container runtime MUST send the seccomp state over the UNIX socket as regular payload.
+The container runtime MUST also send the file descriptor(s) via `SCM_RIGHTS`: the seccomp file descriptor returned by the seccomp syscall and, optionally, the process file descriptor (e.g as returned by `pidfd_open(2)` or by `clone(2)` with the `CLONE_PID` flag).
+The container runtime MAY use several `sendmsg(2)` calls to send the aforementioned data.
+If more than one `sendmsg(2)` is used, the file descriptors MUST be sent only in the first call.
+
+The seccomp state includes the following properties:
+
+* **`ociVersion`** (string, REQUIRED) is version of the Open Container Initiative Runtime Specification with which the seccomp state complies.
+* **`seccompFd`** (int, REQUIRED) is the index of the file descriptor in the `SCM_RIGHTS` array refering to the seccomp notify file descriptor.
+    The value MUST be 0.
+* **`pid`** (int, REQUIRED) is the process ID, as seen by the runtime, on which the seccomp filter is applied (target process).
+* **`pidFd`** (int, OPTIONAL) is the index of the file descriptor in the `SCM_RIGHTS` array refering to the target process file descriptor.
+    If present, this value MUST NOT be zero.
+* **`metadata`** (string, OPTIONAL) is the string set in `listenerMetadata`.
+* **`state`** (map, REQUIRED) is the [state](runtime.md#state) of the container.
+
+Example:
+
+```json
+{
+    "ociVersion": "0.2.0",
+    "seccompFd": 0,
+    "pid": 4422,
+    "pidFd": 1,
```

Comment:
I am not sure about implementing `pidFd` as part of seccomp notifications.

There are cases where pidfd is useful also without seccomp notifications, should we have a separate way for passing the pidfd?

Reply:
I would be fine if the spec define another separate way for passing pidfds.

But I feel it is useful here as well. Note that this `pidFd` is not necessarily referring to the "pid 1" of the container, but in the case of "runc exec", it is the pid that entered in the container. I don't see how we can have a separate way for passing the pidfd in that case.

Reply:
what do you think if we had something not tied to seccomp like?

```
{
    "ociVersion": "0.2.0",
    "fds" : {
       "seccompFd" : 0,
       "pidFd": 1,
       "futureFeatureFd" : 2
    }
    "pid": 4422,
    "state": {
        "ociVersion": "0.2.0",
        "id": "oci-container1",
        "status": "creating",
        "pid": 4422,
        "bundle": "/containers/redis",
        "annotations": {
            "myKey": "myValue"
        }
    }
}
```
Would that work as well?

I am thinking we already have a way to pass the tty fd (with `--console-socket`) and we could perhaps cover it as part of the OCI configuration instead of having it as a CLI option

Reply:
@giuseppe maybe this is a silly question, I think I'm a bit lost. How will the seccomp agent know about the new containers? Do you mean to keep the listenerPath and listenerMatadata in the seccomp section but send this more generic state instead of a seccomp state? In that case, how will be useful for other (non-seccomp) uses? Or do you mean to add fields to the state (not as your example json, as they are not part of the state) and use something like [this](https://github.com/opencontainers/runtime-spec/blob/master/runtime.md#query-state)?

I'm not sure I follow what you mean, or how we can use it from a seccomp agent that doesn't know when runc creates a container or "runc exec" is run in a container, so we have two (or more) pidfds to report: the fd for pid 1 of the container and the fd for the process that run "runc exec".

I'll try to wake up and read it again, just in case :)

Reply:
my suggestion was just about using a map for holding the file descriptors and pass the same metadata.  In this way we could extend it in future, if we'll need any new kind of fd.

Reply:
It sounds good to me to use a `map[string]int` instead of individual fields for seccompFd and pidFd.

If you want to make it more generic, I guess you want to rename `type SeccompState` in `specs-go/state.go`... Do you have an idea of new name?

Reply:
@giuseppe ohh, sorry, I see now! SGTM too, seems like a nice idea. I'll adapt to create a "fds" section in the seccomp state.

Let me know if you want to rename the seccomp state to some other (more generic) thing as alban suggested :)

Reply:
@giuseppe I've moved to the map as you suggested. I couldn't find a good name for [the struct](https://github.com/opencontainers/runtime-spec/pull/1074/files#diff-a1246a93f692debc1727ae5655763e78b2f1662f8cded2473d1403ae2b24d035R37-R48).

What name do you suggest? :)

Reply:
can't really think of anything good to suggest.  Maybe `ContainerState` or `ContainerProcessState`?

Reply:
@giuseppe I just updated the PR to call it `ContainerProcessState`. Please let us know what you think. Thanks!